docs/root/start/sandboxes/single-page-app.rst

```diff
@@ -0,0 +1,497 @@
+and some :ref:`guidance is provided on how to do this <install_sandboxes_single_page_app_step_github_oauth>`.
+
+.. _install_sandboxes_single_page_app_step_hmac:
+
+Step 1: Generate an HMAC secret
+*******************************
+
+Envoy's :ref:`OAuth filter <config_http_filters_oauth>` requires an HMAC secret for encoding credentials.
+
+Change to the ``examples/single-page-app`` directory, and create the HMAC secret.
+
+Replace ``MY_HMAC_SECRET_SEED`` with a phrase of your choosing.
+
+.. code-block:: console
+
+   $ HMAC_SECRET=$(echo "MY_HMAC_SECRET_SEED" | mkpasswd -s)
+   $ export HMAC_SECRET
+   $ envsubst < secrets/hmac-secret.tmpl.yml > secrets/hmac-secret.yml
+
+.. _install_sandboxes_single_page_app_step_start:
+
+Step 2: Start the containers
+****************************
+
+First export ``UID`` to ensure files created by the containers are created with your user id.
+
+Then bring up the docker composition.
+
+.. code-block:: console
+
+    $ pwd
+    envoy/examples/single-page-app
+    $ export UID
+    $ docker compose pull
+    $ docker compose up --build -d
+    $ docker compose ps
+    NAME                          IMAGE                       COMMAND                                                    SERVICE   CREATED         STATUS         PORTS
+    ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    single-page-app-envoy-1       single-page-app-envoy       "/docker-entrypoint.sh envoy -c /etc/envoy/envoy.yaml ..." envoy     2 minutes ago   Up 2 minutes   0.0.0.0:10000-10001->10000-10001/tcp, :::10000-10001->10000-10001/tcp
+    single-page-app-myhub-1       single-page-app-myhub       "/opt/myhub/app.py"                                        myhub     2 minutes ago   Up 2 minutes   0.0.0.0:7000->7000/tcp, :::7000->7000/tcp
+    single-page-app-myhub-api-1   single-page-app-myhub-api   "/opt/myhub/app.py"                                        myhub-api 2 minutes ago   Up 2 minutes
+    single-page-app-ui-1          single-page-app-ui          "/entrypoint.sh dev.sh"                                    ui        2 minutes ago   Up 2 minutes
+
+.. _install_sandboxes_single_page_app_step_login:
+
+Step 3: Browse to the dev app and login
+***************************************
+
+The development app should now be available on at http://localhost:10001 and provide a login button.
+
+.. image:: /start/sandboxes/_static/spa-login.png
+   :align: center
+
+.. note::
+   The dummy OAuth provider automatically trusts everyone as a hard-coded ``envoydemo`` user and redirects back to the app.
+
+   In a real world scenario the provider would authenticate and authorize the user before proceeding.
+
+When a user clicks ``login`` it initiates the OAuth flow by calling the ``/authorize`` path:
```

Comment:
```suggestion
When a user clicks ``login`` the app initiates the OAuth flow by calling the ``/login`` path:
```

current (ie these) docs are not correct it calls `/login` but it could be anything thats not explicitly excluded 

this requires some better explanation and perhaps defaults cc @mattklein123